### ISSUE TITLE
Tune down missing implementation error to warning

### ISSUE THREAD

**Issue by unional** [palantir/tslint#1763]
### Bug Report

- __TSLint version__: `4.0.1`
- __TypeScript version__: `2.0.10`
- __Running TSLint via__: CLI

Since the plan is depreciating `no-unused-variables` and others at 4.0 and remove at 5.0, it should not cause error at runtime right now:

```sh
        throw new Error(ERROR_MESSAGE);
        ^

Error:
            Could not find implementations for the following rules specified in the configuration:
            no-duplicate-key
no-unreachable
            Try upgrading TSLint and/or ensuring that you have all necessary custom rules installed.
            If TSLint was recently upgraded, you may have old rules configured which need to be cleaned up.
```

Most people uses existing config packages (`tslint-config-standard`, `tslint-config-typings`, `tslint-config-unional`, etc), harsh failing causes many builds to fail unnecessarily.

🌷

**Comment by nchen63** [comment]
This error is shown when the rule doesn't exist, not just when it's deprecated. `no-duplicate-key` was removed.

**Comment by unional** [comment]
According to this: https://github.com/palantir/tslint/issues/1481#issue-171019060
It should be just deprecated.

**Comment by nchen63** [comment]
`no-unused-variable` is deprecated. Your error above is not for that rule

**Comment by unional** [comment]
I see. You are correct. However, those two are removed in 4.0 before going into deprecation state. So the change is still significant and affects a lot of people.

In that case, I would suggest keeping those two in depreciation state just like `no-unused-variables` and remove in 5.0.

The few configs I know of are all updated so it may be ok. However there are probably other configs out there used by a lot of people who would be impacted by this.

**Comment by adidahiya** [comment]
Thought about this for a bit and I think @unional is right; we should probably tune down the error to just be a `console.error` and not a thrown exception.

I don't think we need to keep the rules around (might as well make the breaking changes in the major version bump while we can). However, since users of external configs often don't control those configs, our current approach blocks them from upgrading TSLint. In [`ruleLoader.ts`](https://github.com/palantir/tslint/blob/51e399084e9f8cf9bca97f91404a7d342178fc63/src/ruleLoader.ts#L43), we throw errors even for rules that are _disabled_, so end users are not able to escape the error until the config maintainer removes the rule.
